### PR TITLE
[Fuzzing] disable signature verification when fuzzing

### PR DESCRIPTION
**Pull request by snnw**
Disable signature verification when building with `-DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION`.

### CODE REVIEW

**Comment by hanno-arm** [comment]
@snnw @Patater Has there been some discussion / decision in the team as to whether we want to introduce fuzzing-related options into production code? My initial feeling would be that this should be restricted to test-specific branches as it comes at the danger of weakening security by accidental misconfiguration. In case we do want to include such primitives in production code, I would suggest we add very explicit warnings like [for testing `NULL` entropy](https://github.com/ARMmbed/mbedtls/blob/development/library/entropy.c#L30).

@snnw Could you please also name a second reviewer for this? I think our policy is to have at least two reviewers for any PR.

**Comment by Patater** [comment]
If this is the change we want to make, this implementation looks fine to me.

Fuzzing configuration is likely to require more than just one option, so this requires more thought.

File: library/pk.c [hunk 1]
```diff
@@ -226,6 +226,10 @@ int mbedtls_pk_verify_ext( mbedtls_pk_type_t type, const void *options,
         if( sig_len < mbedtls_pk_get_len( ctx ) )
             return( MBEDTLS_ERR_RSA_VERIFY_FAILED );
 
+#if defined(FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION)
```
Comment (gilles-peskine-arm) [inline library/pk.c]: To make it more unlikely that someone will activate this accidentally, I suggest a check like

    #if FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION == 0x4d4f52454d41471c

**Comment by gilles-peskine-arm** [comment]
If we think there's too much risk of activating fuzzing mode in production, we can make it more difficult, for example:

* Only activate fuzzing mode for certain built-in keys or server names, if the fuzzing is always done with these keys/names.
* Add a big delay or an `exit()` somewhere in a place where we wouldn't want the fuzzer to continue but a normal application would — whether we can do that depends on how thoroughly we fuzz.
